### ISSUE TITLE
Unlimited history

### ISSUE THREAD

**Issue by CyberShadow** [Keruspe/GPaste#155]
Hi,

The Windows open-source clipboard manager [Ditto](http://ditto-cp.sourceforge.net/) has an unique feature which I've yet to find in any Linux clipboard manager: unlimited history.

That is, there can be no limit to the number of stored entries. All entries are saved in an SQLite database, which is indexed with SQLite's excellent full text indexing. This allows incrementally searching through hundreds of thousands of clipboard entries in seconds.

I have found this to be incredibly convenient, as it means that I can always cut instead of delete anything, with the peace of mind that I can always recover it later. It also allows me to quickly paste in code snippets written any amount of time ago.

Would this be a viable feature request for this project?

**Comment by Keruspe** [comment]
I guess it could be doable, would require lot of work though and I don't think I'll manage to find that much time for 3.22, but I'll further study the idea and will see if I can make more things modular in preparation for such a feature.

**Comment by Keruspe** [comment]
Fwiw, I just added an abstraction level regarding the storage backend used. The only available for now is regular file, but this opens the door to things like sqlite in the future.

**Comment by kapouer** [comment]
The future has happened :) Do you think this feature could make it ?

**Comment by Keruspe** [comment]
Yes, it's definitely on the list of items I want "ASAP" (including for my own needs). I want to finish the gtk4 port before though.

**Comment by Keruspe** [comment]
I wouldn't recommend this with the file backend, but you should be able to do that with the sqlite backend by setting a very huge max size which will make it virtually unlimited, once 51 is released (soon)
